src/NumericalAlgorithms/LinearSolver/Gmres/InitializeElement.hpp

```diff
@@ -47,49 +52,93 @@ struct InitializeElement {
+                    Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& array_index, const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    db::mutate<operand_tag>(
+        make_not_null(&box),
+        [](const gsl::not_null<db::item_type<operand_tag>*> operand,
+           const db::item_type<source_tag>& source,
+           const db::item_type<operator_applied_to_fields_tag>&
+               operator_applied_to_fields) noexcept {
+          *operand = source - operator_applied_to_fields;
+        },
+        get<source_tag>(box), get<operator_applied_to_fields_tag>(box));
+    const auto& operand = get<operand_tag>(box);
 
     Parallel::contribute_to_reduction<
         gmres_detail::InitializeResidualMagnitude<ParallelComponent>>(
         Parallel::ReductionData<
             Parallel::ReductionDatum<double, funcl::Plus<>, funcl::Sqrt<>>>{
-            inner_product(q, q)},
+            inner_product(operand, operand)},
         Parallel::get_parallel_component<ParallelComponent>(cache)[array_index],
         Parallel::get_parallel_component<ResidualMonitor<Metavariables>>(
             cache));
 
     db::item_type<initial_fields_tag> x0(get<fields_tag>(box));
     db::item_type<basis_history_tag> basis_history{};
 
-    return db::create_from<db::RemoveTags<>, simple_tags, compute_tags>(
-        std::move(box),
-        // We have not started iterating yet, so the _next_ iteration ID is
-        // zero. We initialize the current iteration id such that it advances to
-        // zero by the increment operator.
-        std::numeric_limits<size_t>::max(), size_t{0}, std::move(x0),
-        db::item_type<orthogonalization_iteration_id_tag>{0},
-        std::move(basis_history),
-        db::item_type<LinearSolver::Tags::HasConverged>{});
+    using compute_tags = db::AddComputeTags<
+        ::Tags::NextCompute<LinearSolver::Tags::IterationId>>;
+    return std::make_tuple(
+        ::Initialization::merge_into_databox<
+            InitializeElement,
+            db::AddSimpleTags<
+                LinearSolver::Tags::IterationId, initial_fields_tag,
+                orthogonalization_iteration_id_tag, basis_history_tag,
+                LinearSolver::Tags::HasConverged>,
+            compute_tags, MergePolicy>(
+            std::move(box),
+            // We have not started iterating yet, so we initialize the current
+            // iteration ID such that the _next_ iteration ID is zero.
+            std::numeric_limits<size_t>::max(), std::move(x0),
+            db::item_type<orthogonalization_iteration_id_tag>{0},
+            std::move(basis_history),
+            db::item_type<LinearSolver::Tags::HasConverged>{}),
+        // Terminate algorithm for now. The reduction will be broadcast to the
+        // next action which is responsible for restarting the algorithm.
+        true);
```

Comment:
same pattern as previous comment about algorithm termination, so if changed above, similar changes should be applied here.